src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -9055,9 +9055,23 @@ private ExpressionSyntax ParseSubExpressionCore(Precedence precedence)
             {
                 var questionToken = this.EatToken();
                 var colonLeft = this.ParsePossibleRefExpression();
-                var colon = this.EatToken(SyntaxKind.ColonToken);
-                var colonRight = this.ParsePossibleRefExpression();
-                leftOperand = _syntaxFactory.ConditionalExpression(leftOperand, questionToken, colonLeft, colon, colonRight);
+                if (this.CurrentToken.Kind == SyntaxKind.EndOfFileToken && this.lexer.FollowedByColon)
+                {
+                    // We have an interpolated string with an interpolation that contains a conditional expression.
+                    // Unfortunately, the precedence demands that the colon is considered part of the interpolated
+                    // string. Without this code, the compiler would complain about a missing colon, and point
+                    // to the colon token that is actually there. We need to give a better error message.
+                    var colon = SyntaxFactory.MissingToken(SyntaxKind.ColonToken);
+                    var colonRight = _syntaxFactory.IdentifierName(SyntaxFactory.MissingToken(SyntaxKind.IdentifierToken));
+                    leftOperand = _syntaxFactory.ConditionalExpression(leftOperand, questionToken, colonLeft, colon, colonRight);
+                    leftOperand = this.AddError(leftOperand, ErrorCode.ERR_ConditionalInInterpolation);
```

Comment:
>leftOperand [](start = 48, length = 11)

It might be better to attach this error to the ```questionToken```.

Reply:
Associating it with the entire (partial) conditional expression looks nice in practice. See tests.

Reply:
> Associating it with the entire (partial) conditional expression looks nice in practice. See tests.

I actually did look at the test before making the suggestion and it looked confusing to me because only a portion of a conditional expression was squiggled.

Reply:
Yes, that's what the error is trying to tell you. There is only a portion of a conditional expression present.